pkg/apis/maistra/conversion/extension_providers.go

```diff
@@ -62,6 +62,40 @@ func populateExtensionProvidersValues(in *v2.ControlPlaneSpec, allValues map[str
 				"envoyExtAuthzHttp": values,
 			})
 		}
+		if provider.EnvoyExtAuthzGRPC != nil {
+			config := provider.EnvoyExtAuthzGRPC
+			values := map[string]interface{}{
+				"service": config.Service,
+				"port":    config.Port,
+			}
+			if config.Timeout != nil {
+				values["timeout"] = *config.Timeout
+			}
+			if config.FailOpen != nil {
+				values["failOpen"] = *config.FailOpen
+			}
+			if config.StatusOnError != nil {
+				values["statusOnError"] = *config.StatusOnError
+			}
+
+			if config.IncludeRequestBodyInCheck != nil {
+				includeRequestBodyInCheckValues := map[string]interface{}{}
+				if config.IncludeRequestBodyInCheck.MaxRequestBytes != nil {
+					includeRequestBodyInCheckValues["maxRequestBytes"] = *config.IncludeRequestBodyInCheck.MaxRequestBytes
+				}
+				if config.IncludeRequestBodyInCheck.AllowPartialMessage != nil {
+					includeRequestBodyInCheckValues["allowPartialMessage"] = *config.IncludeRequestBodyInCheck.AllowPartialMessage
+				}
+				if config.IncludeRequestBodyInCheck.PackAsBytes != nil {
+					includeRequestBodyInCheckValues["packAsBytes"] = *config.IncludeRequestBodyInCheck.PackAsBytes
+				}
+				values["includeRequestBodyInCheck"] = includeRequestBodyInCheckValues
+			}
```

Comment:
This is duplicated code. Can you extract it into a function and call it for both HTTP and GRPC?